core/src/lib.rs

```diff
@@ -236,15 +245,12 @@ pub struct ArtifactMetadata {
 }
 
 /// Provides a unique 16-byte ID for a value's type.
+#[cfg(not(feature = "type_uuid"))]
```

Comment:
This would make the behaviour of the feature non-additive. is that really desirable? I'm not comfortable with exporting different traits depending on a feature flag.

Reply:
Okay, good to know.  I figured that the traits had the same signature so it was alright.
I personally found it confusing that there were two traits with the same name accessible by the user when the `type_uuid` flag is enabled because we re-export type_uuid (and it also can mess with auto-import tooling like rust-analyzer).

I'll revert this.

Reply:
Do you think we should rename it perhaps?

Reply:
It works fine the old way for now, I'd rather just get the CI working so I can hack at the functionality parts of the library and understand how it works a bit better.